### PR TITLE
rework internals

### PR DESCRIPTION
**Pull request by pfitzseb**

### CODE REVIEW

**Comment by pfitzseb** [comment]
Ok, tests finally pass on travis (although I needed to disable them on Julia 1.1 because of weird `ERROR: LoadError: GitError(Code:ERROR, Class:SSL, failed to initialize mbedTLS entropy pool)` errors).

File: src/DocumentationGenerator.jl [hunk 1]
```diff
@@ -1,515 +1,149 @@
...
-        push!(exs, ex)
-    end
-    if length(exs) == 0
-        throw(ParseError("end of input"))
-    elseif length(exs) == 1
-        return exs[1]
+    @info("$(packagespec.name) specifies docs of type $(type).")
+    out = if type == "hosted"
+        build_hosted_docs(packagespec, buildpath, uri)
+    elseif type == "git-repo"
```
Comment (vdayanand) [inline src/DocumentationGenerator.jl]: Shouldn't this be `gitdir` now?

Comment (pfitzseb) [reply src/DocumentationGenerator.jl]: No, this looks into the [registry](https://github.com/JuliaDocs/DocumentationGeneratorRegistry/blob/master/Registry.toml#L29). Probably makes sense to switch the output doctypes to `git-repo` as well though.

File: src/DocumentationGenerator.jl [hunk 2]
```diff
@@ -1,515 +1,149 @@
...
 
-            verbose && @info("$name completed in $(round(time() - tstart, digits=1)) seconds")
-        catch err
-            @error "Error while running $(name) with timeout." error=err
-        finally
-            errstr, outstr = readstr_buffer.((out_io, err_io))
-            isempty(outstr) || println(io, outstr)
-            isempty(errstr) || println(io, errstr)
+    # make sure registry is updated *before* we start multiple processes that might try that at the same time
+    Pkg.update()
```
Comment (vdayanand) [inline src/DocumentationGenerator.jl]: Should this be `Pkg.Registry.update()`, since we don't want packages in the active project?

Comment (pfitzseb) [reply src/DocumentationGenerator.jl]: Yes, good catch.

**Pull request merged**